### PR TITLE
Fix scale transparency image in legend

### PR DESCRIPTION
**Pull request by sbrunner**
Fix: #1056

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java [hunk 1]
```diff
@@ -230,7 +230,9 @@ private BufferedImage scaleToMaxWidth(final BufferedImage image, final double sc
         final double factor = this.maxWidth / scaleFactor / image.getWidth();
         final BufferedImage result = new BufferedImage(
                 (int) Math.round(image.getWidth() * factor),
-                (int) Math.round(image.getHeight() * factor), image.getType());
+                (int) Math.round(image.getHeight() * factor),
+                BufferedImage.TYPE_4BYTE_ABGR
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: After that we will store the image as png, then it shouldn't have a big impact...
Or should I use
`image.getType() == BufferedImage.TYPE_BYTE_INDEXED ? BufferedImage.TYPE_4BYTE_ABGR : image.getType()` ?

[the remark this answers is not in the document]
Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: A bit less frightening, yes. If it works for you, I find this solution less dangerous.

[the remark this answers is not in the document]
Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: OK, done :-)

[the remark this answers is not in the document]
Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: I also add TYPE_BYTE_BINARY ...

[the remark this answers is not in the document]
Comment (jwkaltz) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: Not sure if this is relevant, but:

please make sure the "disabling transparency" from https://github.com/mapfish/mapfish-print/pull/939 is not impacted by this.

[the remark this answers is not in the document]
Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: @jwkaltz For me the pdf-a (#939) didn't work's with legends...

[the remark this answers is not in the document]
Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: The pull request affect the scalebar and the north arrow, but not the legend.
On with project is it used? is it possible to test it?

[the remark this answers is not in the document]
Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java]: I just merge it, I let you to test it :-)

**Pull request merged**